asyncapi-cli/src/main/java/io/ballerina/asyncapi/cmd/websockets/AsyncAPIToBallerinaGenerator.java

```diff
@@ -0,0 +1,250 @@
+import io.ballerina.asyncapi.websocketscore.generators.schema.BallerinaTypesGenerator;
+import io.ballerina.asyncapi.websocketscore.model.GenSrcFile;
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.compiler.syntax.tree.TypeDefinitionNode;
+import org.ballerinalang.formatter.core.Formatter;
+import org.ballerinalang.formatter.core.FormatterException;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+import static io.ballerina.asyncapi.cmd.websockets.CmdConstants.CLIENT_FILE_NAME;
+import static io.ballerina.asyncapi.cmd.websockets.CmdConstants.CONFIG_FILE_NAME;
+import static io.ballerina.asyncapi.cmd.websockets.CmdConstants.DEFAULT_CLIENT_PKG;
+import static io.ballerina.asyncapi.cmd.websockets.CmdConstants.GenType.GEN_CLIENT;
+import static io.ballerina.asyncapi.cmd.websockets.CmdConstants.GenType.GEN_SERVICE;
+import static io.ballerina.asyncapi.cmd.websockets.CmdConstants.TEST_DIR;
+import static io.ballerina.asyncapi.cmd.websockets.CmdConstants.TEST_FILE_NAME;
+import static io.ballerina.asyncapi.cmd.websockets.CmdConstants.TYPE_FILE_NAME;
+import static io.ballerina.asyncapi.cmd.websockets.CmdUtils.setGeneratedFileName;
+import static io.ballerina.asyncapi.websocketscore.GeneratorConstants.ASYNCAPI_PATH_SEPARATOR;
+import static io.ballerina.asyncapi.websocketscore.GeneratorConstants.UTIL_FILE_NAME;
+
+/**
+ * This class generates Ballerina Websocket client for a provided AsyncAPI definition.
+ *
+ */
+public class AsyncAPIToBallerinaGenerator {
+    private static final PrintStream outStream = System.err;
+    private String srcPackage;
+    private String licenseHeader = "";
+    private boolean includeTestFiles;
+
+    /**
+     * Generates ballerina websocket client for provided Async API Definition in {@code definitionPath}.
+     * Generated source will be written to a ballerina module at {@code outPath}
+     * Method can be use for generating Ballerina websocket clients.
+     *
+     * @param definitionPath Input Async Api Definition file path
+     * @param outPath        Destination file path to save generated client files including types.bal, utils.bal
+     *                       If not provided {@code definitionPath} will be used as the default destination path
+     * @throws IOException   when file operations fail
+     * @throws BallerinaAsyncApiExceptionWs when code generator fails
+     */
+    public void generateClient(String definitionPath, String outPath)
+            throws IOException, BallerinaAsyncApiExceptionWs, FormatterException {
+        Path srcPath = Paths.get(outPath);
+        Path implPath = CodegenUtils.getImplPath(srcPackage, srcPath);
```

Comment:
when we generate the client, do we check whether it executes within the ballerina package or not ? If not IMO this function call is not necessary.

Reply:
+1, I refactored this in https://github.com/ballerina-platform/asyncapi-tools/pull/119/commits/fb2307ec6eb0833df6feb343c2a9f5aa77455af0